### DISCUSSION TITLE
Unable to Install FastAPI

### DISCUSSION THREAD

**Discussion by wpcodevo** [fastapi/fastapi#6491] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
I can't install fastapi
```

### Description

I created an empty project and created a virtual environment using `py -3 -m venv venv`. After that I activated the virtual environment and started installing fastapi but I was running into the same error severally.

### Operating System

Windows

### Operating System Details

Windows 11

### FastAPI Version

Can't install on my Windows 11

### Python Version

3.10.1

### Additional Context

`Collecting fastapi[all]
  Using cached fastapi-0.71.0-py3-none-any.whl (51 kB)
Collecting pydantic!=1.7,!=1.7.1,!=1.7.2,!=1.7.3,!=1.8,!=1.8.1,<2.0.0,>=1.6.2
  Using cached pydantic-1.9.0-cp310-cp310-win_amd64.whl (2.1 MB)
Collecting starlette==0.17.1
  Using cached starlette-0.17.1-py3-none-any.whl (58 kB)
Collecting ujson<5.0.0,>=4.0.1
  Using cached ujson-4.3.0-cp310-cp310-win_amd64.whl (44 kB)
Collecting requests<3.0.0,>=2.24.0
  Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
Collecting pyyaml<6.0.0,>=5.3.1
  Using cached PyYAML-5.4.1-cp310-cp310-win_amd64.whl
Collecting email_validator<2.0.0,>=1.1.1
  Using cached email_validator-1.1.3-py2.py3-none-any.whl (18 kB)
Collecting itsdangerous<3.0.0,>=1.1.0
  Using cached itsdangerous-2.0.1-py3-none-any.whl (18 kB)
Collecting jinja2<4.0.0,>=2.11.2
  Using cached Jinja2-3.0.3-py3-none-any.whl (133 kB)
Collecting python-multipart<0.0.6,>=0.0.5
  Using cached python-multipart-0.0.5.tar.gz (32 kB)
Collecting orjson<4.0.0,>=3.2.1
  Using cached orjson-3.6.5-cp310-none-win_amd64.whl (181 kB)
Collecting uvicorn[standard]<0.16.0,>=0.12.0
  Using cached uvicorn-0.15.0-py3-none-any.whl (54 kB)
Collecting anyio<4,>=3.0.0
  Using cached anyio-3.4.0-py3-none-any.whl (78 kB)
Collecting sniffio>=1.1
  Using cached sniffio-1.2.0-py3-none-any.whl (10 kB)
Collecting idna>=2.8
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting dnspython>=1.15.0
  Using cached dnspython-2.1.0-py3-none-any.whl (241 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.0.1-cp310-cp310-win_amd64.whl (15 kB)
Collecting typing-extensions>=3.7.4.3
  Using cached typing_extensions-4.0.1-py3-none-any.whl (22 kB)
Collecting six>=1.4.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.10-py3-none-any.whl (39 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.8-py2.py3-none-any.whl (138 kB)
Collecting asgiref>=3.4.0
  Using cached asgiref-3.4.1-py3-none-any.whl (25 kB)
Collecting click>=7.0
  Using cached click-8.0.3-py3-none-any.whl (97 kB)
Collecting h11>=0.8
  Using cached h11-0.12.0-py3-none-any.whl (54 kB)
Collecting httptools==0.2.*
  Using cached httptools-0.2.0.tar.gz (169 kB)
Collecting colorama>=0.4
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting websockets>=9.1
  Using cached websockets-10.1-cp310-cp310-win_amd64.whl (97 kB)
Collecting python-dotenv>=0.13
  Using cached python_dotenv-0.19.2-py2.py3-none-any.whl (17 kB)
Collecting watchgod>=0.6
  Using cached watchgod-0.7-py3-none-any.whl (11 kB)
Using legacy 'setup.py install' for python-multipart, since package 'wheel' is not installed.
Using legacy 'setup.py install' for httptools, since package 'wheel' is not installed.
Installing collected packages: sniffio, idna, colorama, typing-extensions, h11, click, asgiref, anyio, websockets, watchgod, uvicorn, urllib3, starlette, six, pyyaml, python-dotenv, pydantic, MarkupSafe, httptools, dnspython, charset-normalizer, certifi, ujson, requests, python-multipart, orjson, jinja2, itsdangerous, fastapi, email-validator
    Running setup.py install for httptools ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\zidda\Desktop\Learn\python\FASTAPI\venv\Scripts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\zidda\\AppData\\Local\\Temp\\pip-install-gec1csmx\\httptools_4a5b0f863da0494da2983c8bec9efe86\\setup.py'"'"'; __file__='"'"'C:\\Users\\zidda\\AppData\\Local\\Temp\\pip-install-gec1csmx\\httptools_4a5b0f863da0494da2983c8bec9efe86\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\zidda\AppData\Local\Temp\pip-record-3kgm0vl2\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\zidda\Desktop\Learn\python\FASTAPI\venv\include\site\python3.10\httptools'
         cwd: C:\Users\zidda\AppData\Local\Temp\pip-install-gec1csmx\httptools_4a5b0f863da0494da2983c8bec9efe86\
    Complete output (25 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.10
    creating build\lib.win-amd64-3.10\httptools
    copying httptools\_version.py -> build\lib.win-amd64-3.10\httptools
    copying httptools\__init__.py -> build\lib.win-amd64-3.10\httptools
    creating build\lib.win-amd64-3.10\httptools\parser
    copying httptools\parser\errors.py -> build\lib.win-amd64-3.10\httptools\parser
    copying httptools\parser\__init__.py -> build\lib.win-amd64-3.10\httptools\parser
    running egg_info
    writing httptools.egg-info\PKG-INFO
    writing dependency_links to httptools.egg-info\dependency_links.txt
    writing requirements to httptools.egg-info\requires.txt
    writing top-level names to httptools.egg-info\top_level.txt
    reading manifest file 'httptools.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    adding license file 'LICENSE'
    writing manifest file 'httptools.egg-info\SOURCES.txt'
    copying httptools\parser\parser.c -> build\lib.win-amd64-3.10\httptools\parser
    copying httptools\parser\url_parser.c -> build\lib.win-amd64-3.10\httptools\parser
    running build_ext
    building 'httptools.parser.parser' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\zidda\Desktop\Learn\python\FASTAPI\venv\Scripts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\zidda\\AppData\\Local\\Temp\\pip-install-gec1csmx\\httptools_4a5b0f863da0494d'"'; __file__='"'"'C:\\Users\\zidda\\AppData\\Local\\Temp\\pip-install-gec1csmx\\httptools_4a5b0f863da0494da2983c8bec9efe86\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\zidda\AppData\Local\Temp\pip-record-3kgm0vl2\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\zidda\Desktop\Learn\python\FASTAPI\venv\include\site\python3.10\httptools' Check the logs for full command output.
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the 'C:\Users\zidda\Desktop\Learn\python\FASTAPI\venv\Scripts\python.exe -m pip install --upgrade pip' command.`

**Comment by nleroy917** [comment, marked as answer]
The error:
```
running build_ext building 'httptools.parser.parser' extension error: 
Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools":
https://visualstudio.microsoft.com/visual-cpp-build-tools/
```
I'd suggest visiting there and trying to install those.

**Comment by wpcodevo** [comment]
Thanks the issue has been solved
